test/Microsoft.ML.Tests/Transformers/NormalizerTests.cs

```diff
@@ -886,5 +887,93 @@ void TestNormalizeLogMeanVarianceFixZeroVec()
             Assert.Equal(0f, transformedDataArray[2].Features[1]);
             Assert.Equal(0f, transformedDataArray[2].Features[4]);
         }
+
+        [Fact]
+        public void TestNormalizeBackCompatibility2()
+        {
+            // Tests backward compatibility with a normalizing transformer
+            // whose version is "verWrittenCur: 0x00010001"
+
+            string dataPath = GetDataPath(TestDatasets.iris.trainFilename);
+
+            var loader = new TextLoader(Env, new TextLoader.Options
+            {
+                Columns = new[] {
+                    new TextLoader.Column("float1", DataKind.Single, 1),
+                    new TextLoader.Column("float4", DataKind.Single, new[]{new TextLoader.Range(1, 4) }),
+                    new TextLoader.Column("double1", DataKind.Double, 1),
+                    new TextLoader.Column("double4", DataKind.Double, new[]{new TextLoader.Range(1, 4) }),
+                    new TextLoader.Column("int1", DataKind.Int32, 0),
+                    new TextLoader.Column("float0", DataKind.Single, new[]{ new TextLoader.Range { Min = 1, VariableEnd = true } }),
+                },
+                HasHeader = true
+            }, new MultiFileSource(dataPath));
+
+            var data = loader.Load(dataPath);
+
+            var modelPath = Path.Combine("TestModels", "normalizer_verwrit-00010001.zip");
+            var normalizer = ML.Model.Load(modelPath, out var schema);
+
+            var outputPath = GetOutputPath("NormalizerEstimator", "normalized2.tsv");
+            using (var ch = Env.Start("save"))
+            {
+                var saver = new TextSaver(Env, new TextSaver.Arguments { Silent = true });
+                using (var fs = File.Create(outputPath))
+                {
+                    var transformedData = normalizer.Transform(data);
+                    var dataView = ML.Transforms.DropColumns(new[] { "float0" }).Fit(transformedData).Transform(transformedData);
```

Comment:
Why are you dropping `float0` here?

Reply:
I based this test on the [NormalizerWorkout test](https://github.com/dotnet/machinelearning/blob/461b7d38c06d64e53c76e29770665c4b0bb44e72/test/Microsoft.ML.Tests/Transformers/NormalizerTests.cs#L30), where that column is dropped so to be able to compare the transformed data with the file on `normalized.tsv` (which is also done in this test). I've just realized that the `float0 `column is created on that test to work with some things that I don't do in this test; i.e. the normalizer transformer I load from disk doesn't actually do anything with that column.

So I've fixed this by simply removing the part of the `TextLoader `where `float0 `is created, and removed this line of code you've referred since its no longer needed to drop it.

---
In reply to: [334561513](https://github.com/dotnet/machinelearning/pull/4321#discussion_r334561513) [](ancestors = 334561513)